### DISCUSSION TITLE
Desktop version functionality in command line version

### DISCUSSION THREAD

**Discussion by kirnerlukas** [jyrkioraskari/IFCtoLBD#42] [Q&A]
Hi Everyone,

I started to use this converter a while ago for projects in the domain of robotics as the inclusion of the bot concepts on top of ifcOWL has a lot of benefits for my use-cases. For proof of concept I used the desktop Version as the main part of my developments happen after conversion and the source IFC did not change over time.

Now I want to integrate it in my infrastructure which works perfectly fine so far. I just wonder how I can use some of the desktop versions functionality like "seperate file" or keeping the ifcOWL via the CLI Version. Especially keeping the ifcOWL and having the 'owl:sameAs' in the LBD ttl is crucial for me.

How could I achieve this?

**Comment by jyrkioraskari** [comment]
Hi Lukas,

Thank you for the excellent comment. That option is not there yet, but it is easy to add. It is just an argument mapping to the converter core. I added this to the issue list so that It will be handled in the next release. 

The fix will be here:
https://github.com/jyrkioraskari/IFCtoLBD/blob/master/IFCtoLBD/src/main/java/org/linkedbuildingdata/ifc2lbd/IFCtoLBDConverter.java

Under row: 329. The command line option uses some default options. That needs to be changed. If it's urgent, I can fix this pretty immediately.
 
There are three other options also:  
- the RESTful API interface 
- the Python library implementation
- and, if you use Java, the IFCtoLBD project can be used as a library to call the conversion function. The desktop app calls the conversion library like this:
 https://github.com/jyrkioraskari/IFCtoLBD/blob/master/IFCtoLBD_Desktop/src/main/java/org/linkedbuildingdata/ifc2lbd/ConversionThread.java

Kind regards,

Jyrki
